File: sdk-java/src/test/java/io/superdurable/dex/iwfcompat/ModelInputFlow.java
/*
 * Copyright (c) 2026 Super Durable, Inc.
 *
 * Licensed under the Super Durable Source License 1.0.
 * You may not use this file except in compliance with the License.
 * See the LICENSE file in the repository root.
 *
 * SPDX-License-Identifier: LicenseRef-Super-Durable-1.0
 */

package io.superdurable.dex.iwfcompat;

import io.superdurable.dex.Context;
import io.superdurable.dex.Flow;
import io.superdurable.dex.Step;
import io.superdurable.dex.StepDef;
import io.superdurable.dex.StepDecision;

import java.util.Collections;
import java.util.List;

final class ModelInputFlow implements Flow<IwfFlows.ModelInput> {
    private final Step<IwfFlows.ModelInput> start = new Step<IwfFlows.ModelInput>() {
        @Override
        public Class<IwfFlows.ModelInput> getInputType() {
            return IwfFlows.ModelInput.class;
        }

        @Override
        public StepDecision execute(final Context context, final IwfFlows.ModelInput input) {
            return StepDecision.gracefulComplete(input.value);
        }
    };

    @Override
    public List<StepDef> getSteps() {
        return Collections.singletonList(StepDef.startStep(start));
    }
}

File: sdk-typescript/test/iwfcompat/model_input_flow.ts
// Portions of this file are derived from indeedeng/iwf-java-sdk.
// Those portions are licensed under the Apache License, Version 2.0.
// See LICENSES/Apache-2.0.txt and LEGACY_NOTICES.md.
//
// Modifications Copyright (c) 2026 Super Durable, Inc.
//
// Modifications are licensed under the Super Durable Source License 1.0.
// Third-Party Materials remain under the Apache License, Version 2.0.
// See LICENSE and LEGACY_NOTICES.md.

import {
  StepDef,
  gracefulComplete,
  type Context,
  type Flow,
  type Step,
  type StepDecision,
} from "../../src/index.js";

import { modelInputCodec, type ModelInput } from "./shared.js";

class ModelInputStep implements Step<ModelInput> {
  public readonly inputCodec = modelInputCodec;

  public getStepType(): string {
    return "ModelInputStep";
  }

  public execute(_context: Context, input: ModelInput): StepDecision {
    return gracefulComplete(input.value);
  }
}

export class ModelInputFlow implements Flow<ModelInput> {
  private readonly start = new ModelInputStep();

  public getFlowType(): string {
    return "ModelInputFlow";
  }

  public getSteps() {
    return [StepDef.startStep(this.start)];
  }
}
